library/dd-sdk-android-ndk/src/main/kotlin/com/datadog/android/ndk/NdkCrashReportsFeature.kt

```diff
@@ -4,73 +4,66 @@
  * Copyright 2016-Present Datadog, Inc.
  */
 
-@file:Suppress("DEPRECATION")
-
 package com.datadog.android.ndk
 
-import android.util.Log
-import com.datadog.android.plugin.DatadogContext
-import com.datadog.android.plugin.DatadogPlugin
-import com.datadog.android.plugin.DatadogPluginConfig
+import android.content.Context
+import com.datadog.android.core.internal.utils.internalLogger
 import com.datadog.android.privacy.TrackingConsent
+import com.datadog.android.privacy.TrackingConsentProviderCallback
+import com.datadog.android.v2.api.EnvironmentProvider
+import com.datadog.android.v2.api.Feature
+import com.datadog.android.v2.api.InternalLogger
+import com.datadog.android.v2.api.SdkCore
 import java.io.File
 import java.lang.NullPointerException
 
 /**
- * An implementation of the [DatadogPlugin] which will allow to intercept and report the
+ * An implementation of the [Feature] which will allow to intercept and report the
  * NDK crashes to our logs dashboard.
  */
-@SuppressWarnings("TooGenericExceptionCaught")
-class NdkCrashReportsPlugin : DatadogPlugin {
+class NdkCrashReportsFeature : Feature, TrackingConsentProviderCallback {
     private var nativeLibraryLoaded = false
 
-    init {
-        var exception: Throwable? = null
-        try {
-            System.loadLibrary("datadog-native-lib")
-            nativeLibraryLoaded = true
-        } catch (e: SecurityException) {
-            exception = e
-        } catch (e: NullPointerException) {
-            exception = e
-        } catch (e: UnsatisfiedLinkError) {
-            exception = e
-        }
-        exception?.let {
-            Log.e(TAG, ERROR_LOADING_NATIVE_MESSAGE, exception)
-        }
-    }
+    override val name: String = "ndk-crash-reporting"
 
-    // region Plugin
-    override fun register(config: DatadogPluginConfig) {
+    // region Feature
+    override fun onInitialize(
+        sdkCore: SdkCore,
+        appContext: Context,
+        environmentProvider: EnvironmentProvider
+    ) {
+        loadNativeLibrary()
         if (!nativeLibraryLoaded) {
             return
         }
         val ndkCrashesDirs = File(
-            config.storageDir,
+            environmentProvider.rootStorageDir,
             NDK_CRASH_REPORTS_FOLDER
         )
         try {
             ndkCrashesDirs.mkdirs()
         } catch (e: SecurityException) {
-            Log.e("Datadog", "Unable to create NDK Crash Report folder $ndkCrashesDirs")
+            internalLogger.log(
+                InternalLogger.Level.ERROR,
+                InternalLogger.Target.USER,
+                "Unable to create NDK Crash Report folder $ndkCrashesDirs",
+                e
+            )
             return
         }
         registerSignalHandler(
             ndkCrashesDirs.absolutePath,
-            consentToInt(config.trackingConsent)
+            consentToInt(environmentProvider.trackingConsent)
```

Comment:
I don't remember how we were doing this before but have you considered the case when the tracking consent is changed ?

Reply:
yes, `onConsentUpdated` callback is still there https://github.com/DataDog/dd-sdk-android/pull/1262/files#diff-f688390d6a60752a35fe77982ed01029b262b098deaf5bc55b162cc68ecabd7dR71